src/components/file-viewer/file-viewer.tsx

```diff
@@ -0,0 +1,225 @@
+import { Component, Element, h, Prop } from '@stencil/core';
+
+/**
+ * This is a smart component that detects most common file types such as
+ * image, audio, video, or text files,
+ * and renders them properly using native HTML5 elements.
+ *
+ * :::note
+ * Image files will always be contained in their containers, which means
+ * they automatically increases or decreases in size to fill the box
+ * whilst preserving thier aspect-ratio.
+ *
+ * Text and PDF files will also always respect the width and height of the
+ * container in which the `limel-file-viewer` is loaded.
+ * :::
+ *
+ * For some file types such as text and images, the component will display a
+ * download button and a button to open the file in a new browser tab.
+ * This will allow users to preview the file in a fullscreen more with the
+ * browser and take advantage of for example native zooming and panning
+ * functionalities.
+ *
+ *:::note
+ * The download button will not work in this documentation, since example files
+ * are not hosted in the same domain.
+ *:::
+ *
+ * @exampleComponent limel-example-file-viewer
+ * @exampleComponent limel-example-file-viewer-with-picker
+ */
+
+@Component({
+    tag: 'limel-file-viewer',
+    shadow: true,
+    styleUrl: 'file-viewer.scss',
+})
+export class FileViewer {
+    /**
+     * Link to the file
+     */
+    @Prop({ reflect: true })
+    public url: string;
+
+    /**
+     * Alternative text for assistive technologies and screen readers, displayed for image files
+     */
+    @Prop({ reflect: true })
+    public alt: string;
+
+    /**
+     * The file type
+     */
+    @Prop({ reflect: true })
+    public type: string;
```

Comment:
This is the MIME type? We should clarify that in the documentation :slightly_smiling_face:

Reply:
yes

Reply:
@jgroth Either rename it to `contentType` (because https://github.com/Lundalogik/lime-elements/blob/ba612ac15dc4722b14880cd1a55e364009765996/src/components/file/file.tsx#L223) or keep the property name. We could steal this text that documents the property https://developer.mozilla.org/en-US/docs/Web/API/File/type#value or just use the same text as on FileInfo (https://github.com/Lundalogik/lime-elements/blob/fd23f9a530ed689f8c9509b84000db0159ab1971/src/components/file/file.types.ts#L17)